Sources/Rendering/Core/Camera/index.js

```diff
@@ -417,6 +417,10 @@ function vtkCamera(publicAPI, model) {
     // invert to get hmd -> world
     mat4.invert(viewMatrix, viewMatrix);
 
+    publicAPI.computeViewParametersFromViewMatrix();
+  };
+
+  publicAPI.computeViewParametersFromViewMatrix = () => {
     // then extract the params position, orientation
     // push 0,0,0 through to get a translation
     vec3.transformMat4(tmpvec1, origin, viewMatrix);
```

Comment:
viewMatrix is not defined here? You may need to store / retrieve it from the model...

Reply:
Same as above.

The idea is to have a function scope viewMatrix as well as a model scope one. When an external application sets the viewMatrix explicitly, it is assigned to the model scope and used whenever `getViewMatrix` is called.

Reply:
Should the publicAPI.computeViewParametersFromViewMatrix function put on the public API?

Reply:
Just don't bind that function on publicAPI and that will be fine. that way the function will be private inside that scope.